0.1

MrSerge01 opened the pull request:
# 0.1
## Add
### Commands
- [x] /moderation
  - [x] lock
  - [x] unlock
- [x] /leaderboard (credit: koslz)

### Settings
- [x] Add settings' descriptions to explain what they do and what has to be set
- [x] moderation.logMessages (BOOL) - whether or not should the bot log message changes
- [x] levelling
  - [x] .blockChannels (STRING) - channels where the levelling system won't count the messages in
  - [x] .XPGain (INTEGER) - sets the XP gain
  - [x] .cooldown (INTEGER) - sets the cooldown between messages which will add xp (thanku koslz)
  - [x] .difficulty (INTEGER) - sets how much more xp one has to get in order to level up
- [x] news.editOriginalMessage (BOOL) - whether or not the bot should edit the original news post or send a new one instead
- [x] welcome
  - [x] .text (STRING) - sets the text of the welcome embed
  - [x] .goodbyeText (STRING) - sets the text of the goodbye embed
  - [x] .channel (STRING) - sets the channel ID where the welcome messages will be sent
  - [x] .dm (BOOL) - whether or not the bot should DM the user when they join the server (default = true)

## Improve
- [x] Permissions that the bot uses (in order to stop forcing ppl to give us admin perms)
- [x] Make most /moderation commands use modEmbed
- [x] Put default values for most settings
- [x] Add tempban and hackban functionality to the existing ban command
- [x] Send an embed to the log channel when the bot unbans a user automatically (KZ IS CARRYING?? sadly, not)
- [x] Use numbers instead of UUID as case ID
- [x] /settings command should use the default values when there are none set

## Error handling
- [x] More checks to error handle buttons
  - [x] Make buttons auto disable themselves after the time period passes
  - [x] Make buttons unclickable for people that haven't sent the message

## Fix
- [x] /user errors (credit: Golem)
- [x] "interaction has been acknowledged" and "interaction has already been replied to" errors
- [x] Cooldowns not working
- [x] Levelling up multiple times is a nonexistent term for the bot
- [x] Levelling up is a nonexistent term for the bot

# Road to Core 1.0 (#33)
## Add
### Features
- Simple multi-currency/scoring system
- Per-user message clearing
- Channel muting (member gets their access to do anything revoked in a single channel)
- Experimental button commands (i.e all user-related actions are buttons in /user, channel-related actions are in /channel, etc.)

### Cases
- Let user sort cases by type of infraction or by user
- Let user edit cases (for example: change duration of a mute, convert ban to tempban and change duration of that)

### Settings
- levelling
  - .addReward (role: ROLE, level: INTEGER) - adds a reward
  - .addMultiplier (multiplier: INTEGER, ID: ROLE[] | CHANNEL[]) - adds a multiplier tied to a role (or a channel)
  - .setLevel (userID: STRING, level: INTEGER) - sets the level of a user to something
  - .XPPerChars (XP: INTEGER, chars: INTEGER) - sets the amount of XP the bot should add per amount of characters written
- news
  - .addCategory (name: STRING, role: ROLE) - adds a category
  - .removeCategory (INTEGER) - removes a category by the ID
- command.toggle (COMMAND) - enables (or disables) a command from the list of available commands
- welcome
 - .joinText (STRING (title, description)) - sets the title and description of the welcome embed 
 - .goodbyeText (STRING (title, description)) - sets the title and description of the goodbye embed
- role
  - retain (keep: BOOL, except: ROLE[]) - when a user rejoins a server, their roles retain except for ones in `except`

## Improve
### Settings
- /settings command embeds
- Type errors for settings (for example, when you provide "auirfhwauirhaiur" as an argument instead of "true" or "false")
- Types
  - `channel` - lists all channels in the server
  - `role` - lists all roles in the server
  - `user` - lists all users in the server
  - `command` - lists all commands of the bot
- Make the bot send a modal when you set a setting with multiple values
- Setting autocomplete when you get/set a setting

### Not settings
- Logging system (use the audit log to log more stuff)

# For versions beyond
- Automod support

MrSerge01 committed:
unfinished 0.2

MrSerge01 committed:
0.2 -> 0.1 (no dashboard to count as 0.2)

MrSerge01 committed:
excess commands moment

MrSerge01 committed:
okay now this is done

Froxcey committed:
Add sqlite querys

MrSerge01 committed:
started fixing things

MrSerge01 committed:
some more changes

MrSerge01 committed:
goofy

Froxcey commented:
Looking good :D
![Okay](https://media.tenor.com/IwiXYMAWFswAAAAC/anime-nichijou.gif)

MrSerge01 committed:
fixed moderation commands

Co-authored-by: Mart Zielman <ThatFrogDev@users.noreply.github.com>

Froxcey committed:
Fix serverboard query

Froxcey committed:
Fix import

MrSerge01 committed:
re-added welcome/goodbye messages

MrSerge01 committed:
errory serverboard.ts

Froxcey committed:
Improve serverboard handling

Froxcey committed:
Fix serverboard

Froxcey committed:
Fix serverboard

MrSerge01 committed:
prettierrc + errorEmbed edit

MrSerge01 committed:
120 -> 100

Froxcey committed:
Add settings

Froxcey committed:
Merge Goos's patch from 'origin/HEAD' into dev

Froxcey committed:
Improve error

MrSerge01 commented:
nebla is nebling

MrSerge01 committed:
prettified all files + fixed some bugs

MrSerge01 committed:
removing references to the dashboard (0.1-pre doesnt have it)

MrSerge01 committed:
errors updated

MrSerge01 committed:
🐛gy stuff, please fix 😭

dimkauzh commented:
Nice

Froxcey committed:
Remove UPDATE Query

Froxcey committed:
Get news by id

MrSerge01 committed:
small progress

MrSerge01 committed:
slowly fixing... and killing my sanity

MrSerge01 committed:
aaaaaaaaaaaaaaaaa

MrSerge01 committed:
test

SwiftDust committed:
Started at the poll command and some fixes in the code

MrSerge01 committed:
fixed /poll

MrSerge01 committed:
quick fix

SwiftDust committed:
Did some more fixes to /poll, work in progress.

SwiftDust committed:
Added a reaction handler for poll

MrSerge01 committed:
errorEmbed redo (again, part 124) and some improvements

MrSerge01 committed:
finished changing errorEmbed

dimkauzh commented:
Looking even better than better

MrSerge01 committed:
made /server news send actually send news, more error handling

MrSerge01 committed:
finished fixing that one bug

MrSerge01 committed:
removed imageurl

MrSerge01 committed:
send help please

MrSerge01 committed:
trying to fix news

MrSerge01 committed:
fixed removing news

MrSerge01 committed:
more changes woo

SwiftDust committed:
feat: settings--types for welcome/goodbye messages

SwiftDust committed:
Merge branch 'dev' of https://github.com/Chiissu/Nebula into dev

MrSerge01 committed:
finished welcome/goodbye, fixed unban

dimkauzh reviewed (approved):
Looking even better than better than better

MrSerge01 commented:
> Looking even better than better than better

yay

MrSerge01 committed:
message logs

dimkauzh commented:
Message logs :O

SwiftDust committed:
todoitem: added a setting and started on LIST type

MrSerge01 committed:
small commit

dimkauzh commented:
Nebula going good :mechanical_arm:

MrSerge01 committed:
some changes

MrSerge01 committed:
user changes

MrSerge01 committed:
levelling works

MrSerge01 committed:
a yet another small commit

MrSerge01 committed:
crazy

MrSerge01 committed:
implement JSON type and stuff like that

MrSerge01 committed:
suffer time

MrSerge01 committed:
america ya :D

SwiftDust committed:
add descriptions for settings, wip

MrSerge01 committed:
some more cool stuff

Co-authored-by: Mart Zielman <ThatFrogDev@users.noreply.github.com>

MrSerge01 committed:
setLevel works

MrSerge01 committed:
addMultiplier

dimkauzh commented:
Wawawawa

MrSerge01 commented:
> Wawawawa

wawawawa?

dimkauzh commented:
> > Wawawawa
> 
> wawawawa?

Quack quack

MrSerge01 commented:
> > > Wawawawa
> > 
> > 
> > wawawawa?
> 
> Quack quack

# HONK HONK

MrSerge01 commented:
btw sorry for force pushing there was a bad commit that i had to remove :)

MrSerge01 committed:
modEmbed hell begins now

MrSerge01 committed:
modEmbed cooking

MrSerge01 committed:
more modEmbed, more!!!!

Golem642 committed:
i hate ts

Golem642 committed:
user profile and settings fixed, + (un)lock channels

MrSerge01 committed:
small stuff

MrSerge01 committed:
index readded + some more stuf

MrSerge01 committed:
one more thing

dimkauzh commented:
Yooooo progress -------------->

MrSerge01 committed:
some /user changes + even more modEmbed

dimkauzh commented:
What's this bot btw, what does it do?

MrSerge01 committed:
i think modEmbed is finally done

MrSerge01 committed:
button progress (more on this later)

MrSerge01 committed:
guh

MrSerge01 committed:
ok these changes are unfinished as hell lmao

MrSerge01 committed:
it works

MrSerge01 committed:
hell yeah

MrSerge01 committed:
oh god

MrSerge01 committed:
small changes

MrSerge01 committed:
interaction errors FIXED

dimkauzh commented:
Jk

SwiftDust committed:
feat: temporary ban

MrSerge01 committed:
smol

MrSerge01 committed:
inconsistency fixing + one more step towards fixing interaction errors

dimkauzh commented:
Commits 👍

dimkauzh commented:
🍞 Bread 👍

MrSerge01 committed:
fixed a couple more issues

Golem642 committed:
mdoeraition

Golem642 committed:
dm welcome attempt n°1

Golem642 committed:
make it jollier

Golem642 committed:
getting closer to finishing the preview

dimkauzh commented:
Golem grindin 💪

Golem642 committed:
no more red

fixed every error i came across, also added the outsideError check for errorEmbed to check if a member is in the server or not

Golem642 committed:
woops just fixing don't mind me

iakzs committed:
something

to test fr

iakzs committed:
testing fr

real test

iakzs committed:
hmm

yes

iakzs committed:
ok bro

iakzs committed:
updat levelling.ts

i hate vs code why cant it work correctly with githubhasudvasjdfhabd

iakzs committed:
updat leaderboard.ts

iakzs committed:
Merge branch 'dev' of https://github.com/iakzs/Nebula into dev

iakzs commented:
w rizz

MrSerge01 committed:
Merge pull request #1 from iakzs/dev

very epic leaderboard

iakzs committed:
added cooldown

iakzs committed:
better sort + using math.floor

iakzs committed:
Merge branch 'dev' of https://github.com/iakzs/Nebula into dev

iakzs committed:
small

iakzs committed:
Merge branch 'dev' of https://github.com/iakzs/Nebula into dev

iakzs committed:
i forgor

iakzs committed:
Merge branch 'dev' of https://github.com/iakzs/Nebula into dev

MrSerge01 committed:
Merge pull request #2 from iakzs/dev

2nd pull req yayy

MrSerge01 commented:
koslz and golem are doing gods work :pray: :pray:

iakzs commented:
# **im epic**
## real things
i am him
### i forgot
but ok

# what if im not epic
## what would happen
idk, w rizz
### would goos give you free robux
yes

iakzs commented:
just did that to see the markdown lol

MrSerge01 commented:
> just did that to see the markdown lol

you couldve just clicked "preview" in the message box

iakzs committed:
temp ban

iakzs committed:
maybe a fix if an interaction error shows up

iakzs committed:
added expired bans

iakzs committed:
added the expired ban check at the start

iakzs committed:
Merge branch 'dev' of https://github.com/iakzs/Nebula into dev

iakzs committed:
added myself to the about hehe

iakzs committed:
Merge branch 'dev' of https://github.com/iakzs/Nebula into dev

iakzs commented:
> > just did that to see the markdown lol
> 
> you couldve just clicked "preview" in the message box

nuh uh

iakzs committed:
yes

iakzs committed:
Merge branch 'dev' of https://github.com/iakzs/Nebula into dev

Commit:
cool

Commit:
forgot the bun.lockb

iakzs committed:
updat unban scheduler

setTimeout (i think) only uses some memory at time of making the ban and the time of unbanning. but if i (1st idea i had) made it check every 1 minute or something would use more memory. i might be wrong

MrSerge01 committed:
Merge pull request #3 from iakzs/dev

very epic temp ban pull request + fix + added myself to the about command hehe

MrSerge01 committed:
eternal suffering, pt1

MrSerge01 committed:
eternal suffering, pt2

MrSerge01 commented:
pain

dimkauzh commented:
> pain 

pain is gain

ZakaHaceCosas commented:
> > pain
> 
> pain is gain

no pain no gain

MrSerge01 commented:
> > > pain
> > 
> > pain is gain
> 
> no pain no gain

gain pain
take away n from gain you get gay
gay pain
🗿🗿

ZakaHaceCosas commented:
> > > > pain
> > > 
> > > 
> > > pain is gain
> > 
> > 
> > no pain no gain
> 
> gain pain take away n from gain you get gay gay pain 🗿🗿

# gay pain fr
now go coding

iakzs commented:
Me fr

dimkauzh commented:
W bot

iakzs commented:
w rizz

MrSerge01 committed:
eternal suffering, pt3

MrSerge01 committed:
HUH

iakzs reviewed (commented):
i agree

dimkauzh reviewed (changes requested):
Finish it :)

MrSerge01 commented:
wow conflicts

dimkauzh commented:
<img width="547" alt="Screenshot 2024-10-20 at 20 00 10" src="https://github.com/user-attachments/assets/2daacb34-98b7-46cc-b3c4-5adbeaab0258">
copilot says this aint ready so no :(

MrSerge01 committed:
Merge branch 'dev' into dev

iakzs commented:
> copilot says this aint ready so no :(
> 
> 
> 
> <img width="547" alt="Screenshot 2024-10-20 at 20 00 10" src="https://github.com/user-attachments/assets/2daacb34-98b7-46cc-b3c4-5adbeaab0258">
> 
> 

Ermmm no ai 😡😡😡🤓

MrSerge01 committed:
eternal suffering, pt4

MrSerge01 committed:
small emoji update

iakzs reviewed (commented):
i fw all of them

dimkauzh commented:
best bot ever

MrSerge01 committed:
happy halloween (eternal suffering pt5) [fox happy now?]

MrSerge01 committed:
eternal suffering pt6: THE END

MrSerge01 committed:
this is not suffering anymore

MrSerge01 committed:
this is not suffering anymore

dimkauzh reviewed (approved):
As a TS specialist (0 TS projects) I can fully approve these changes. Just joking. All looks good to me mate! Good job team!

dimkauzh committed:
Switch to github actions

dimkauzh committed:
possible macos fix

MrSerge01 committed:
Merge pull request #4 from dimkauzh/dev

Rewrite to Github Actions

Golem642 commented on src/commands/About.ts:
```diff
@@ -0,0 +1,77 @@
+        {
+          name: "🔗 • Links",
+          value:
+            "[GitHub](https://www.github.com/NebulaTheBot) • [YouTube](https://www.youtube.com/@NebulaTheBot) • [Instagram](https://instagram.com/NebulaTheBot) • [Mastodon](https://mastodon.online/@NebulaTheBot@mastodon.social) • [Guilded](https://guilded.gg/Nebula) • [Revolt](https://rvlt.gg/28TS9aXy)"
```
Why tf do we still have an IG 😭

MrSerge01 replied:
lmao i might remov

dimkauzh replied:
No leave insta

dimkauzh replied:
Let it be frfr

MrSerge01 replied:
we'll decide

Golem642 commented on src/commands/Leaderboard.ts:
```diff
@@ -0,0 +1,104 @@
+
+    if (totalPages > 1) {
+      const collector = reply.createMessageComponentCollector({
+        time: 60000
```
Not gonna comment all of them, but i think this still has the problem of not being dependent of the last interaction. Is there a way to do this.time += 10000 when a button is pressed or something?

MrSerge01 replied:
the reply is tied to the interaction, thus the message collector is also tied
there's a new error message to ensure it won't fail
about the time, dunno, i might attempt that

MrSerge01 replied:
done

userandaname reviewed (approved):
certifiably FIRE 🔥 🔥 🔥 

congrats on the hard work you all have done to get here!
i'm excited to see what comes next in ~~nebula~~ sokora's future!

ZakaHaceCosas committed:
(fix) Replace deprecated component

ZakaHaceCosas committed:
(feat) Allow to enable/disable easter eggs (making them global)

Golem642 commented on src/commands/Settings.ts:
```diff
@@ -0,0 +1,131 @@
+    settingsKeys.forEach(key => {
+      const subcommand = new SlashCommandSubcommandBuilder()
+        .setName(key)
+        .setDescription("This subcommand has no description.");
```
We should add a way to get the description instead of having this for every subcommands

MrSerge01 replied:
i have an idea about that

MrSerge01 replied:
optix has fixed that

ZakaHaceCosas replied:
this was resolved 👍

Golem642 commented on src/commands/User.ts:
```diff
@@ -0,0 +1,154 @@
+        },
+        {
+          name: "📒 • Server info",
+          value: serverInfo.join("\n")
```
What if they never joined the server ? (You can use a user id with the option attribute I think)

MrSerge01 replied:
thats a great question, will fix that

Golem642 commented on src/commands/moderation/Ban.ts:
```diff
@@ -0,0 +1,66 @@
+        );
+
+      expiresAt = Date.now() + durationMs;
+      scheduleUnban(interaction.client, guild.id, user.id, interaction.member!.user.id, durationMs);
```
Still untested with negative numbers ?

MrSerge01 replied:
fixed

Golem642 commented on src/commands/moderation/Cases.ts:
```diff
@@ -0,0 +1,92 @@
+  }
+
+  async run(interaction: ChatInputCommandInteraction) {
+    const actionsEmojis: { [key: string]: string } = {
+      WARN: "⚠️",
+      MUTE: "🔇",
+      KICK: "📤",
+      BAN: "🔨"
+    };
```
Maybe group these in a constants file ? Idk if they are used elsewhere

MrSerge01 replied:
they are not

Golem642 commented on src/commands/moderation/Lock.ts:
```diff
@@ -0,0 +1,80 @@
+        .create(guild.id, {
+          SendMessages: false,
+          SendMessagesInThreads: false,
+          CreatePublicThreads: false
```
Create private threads too

Golem642 commented on src/commands/moderation/Unlock.ts:
```diff
@@ -0,0 +1,80 @@
+        .create(guild.id, {
+          SendMessages: null,
+          SendMessagesInThreads: null,
+          CreatePublicThreads: null
```
Same thing

Golem642 commented on src/handlers/commands.ts:
```diff
@@ -1,65 +1,75 @@
+    const commandsPath = join(process.cwd(), "src", "commands");
     const command = new SlashCommandBuilder()
       .setName(name.toLowerCase())
       .setDescription("This command has no description.");
```
Same thing as before, why does it have to be like that ?

MrSerge01 replied:
why does what to be like that?

Golem642 replied:
The description being empty

MrSerge01 replied:
because the commands change that, thats a fallback for when a developer doesnt provide a description to avoid an error

Golem642 reviewed (changes requested):
Might have missed some but there are some last few preps before we can consider this done. Good job anyway everyone 👍

ZakaHaceCosas committed:
(feat) Add descriptions to /settings commands

MrSerge01 committed:
Merge pull request #5 from ZakaHaceCosas/dev

Add description to `/setting` commands and allow to enable/disable easter eggs

Froxcey committed:
Add setup command

MrSerge01 committed:
Merge pull request #6 from Chiissu/setup

Add setup command

Golem642 commented on CONTRIBUTING.md:
```diff
@@ -1,31 +1,24 @@
 ### Running
-- Run `bun i` to install all the modules needed by the bot. 
-- Run `bun start`
+- Run `bun start`.
```
Ermm.. achktually, I use `bun run start` ☝️🤓

MrSerge01 replied:
![image](https://github.com/user-attachments/assets/d118be23-4736-4113-80ae-8442cd481cd2)
sorry not sorry

iakzs commented:
so sigma

MrSerge01 committed:
the suffering is truly never over

ZakaHaceCosas reviewed (approved):
You guys have reasons to be proud of yourselves.

LGTM.

MrSerge01 committed:
quick fix

Sammy99jsp commented:
Yoo, that's sick!

Golem642 commented:
Is this finally Sokover ? 😔

MrSerge01 commented:
> Is this finally Sokover ? 😔

accept the pr first if you don't have any more changes

MrSerge01 committed:
(fix) 60 -> 30s, imageColor fix, user nickname

Golem642 reviewed (approved):
LGTM, a few tweaks here and there could be nice but overall thing is working as intended.
If anyone has any objection/changes to do, I'll include mines alongside

Froxcey commented:
I was here

<img src="https://github.com/Avdan-OS/Compositor/assets/51555391/d0379882-f2dc-42e1-962f-b3f122db656f" alt="vibe" width="60"/>

ThatFrogDev commented:
Hello,

Great job, guys! You all did truly amazing work. I have worked together with the rest of the team to test it, I've seen how the embeds of most commands look and it looks utterly great from first sight. Right now, I don't really have the time to review it. I've started reviewing but this is a hell of a pull request so that requires a lot of effort. Later on I'll publish a few of my less important concerns that can be fixed in hindsight.

Thanks for making this year-long dream finally truth. Hereby I speak the magic words, LGTM. 
Mart
